pkg/rhos/ocpgwdeployer.go

```diff
@@ -0,0 +1,176 @@
+	}
+
+	// Currently, we only support increasing the number of Gateway nodes which could be a valid use-case
+	// to convert a non-HA deployment to an HA deployment. We are not supporting decreasing the Gateway
+	// nodes (for now) as it might impact the datapath if we accidentally delete the active GW node.
+	if numGatewayNodes < input.Gateways {
+		gatewayNodesToDeploy := input.Gateways - numGatewayNodes
+
+		workerNodes, err := d.K8sClient.ListNodesWithLabel("node-role.kubernetes.io/worker")
+		if err != nil {
+			return errors.WithMessagef(err, "failed to list k8s nodes  in project %q", d.projectID)
+		}
+
+		nodes := workerNodes.Items
+		for i := range nodes {
+			alreadyTagged := nodes[i].GetLabels()[submarinerGatewayNodeTag]
+			if alreadyTagged == "true" {
+				continue
+			}
+
+			reporter.Started(fmt.Sprintf("Configuring worker node %q as gateway node", nodes[i].Name))
+
+			err := d.K8sClient.AddGWLabelOnNode(nodes[i].Name)
+			if err != nil {
+				return errors.WithMessagef(err, "failed to label the node %q as gateway node", nodes[i].Name)
+			}
+
+			if err := d.openGatewayPort(groupName, nodes[i].Name, computeClient); err != nil {
+				return errors.WithMessage(err, "failed to open the gateway port")
+			}
+
+			gatewayNodesToDeploy--
+			if gatewayNodesToDeploy <= 0 {
+				reporter.Succeeded("Successfully deployed gateway node")
+				return nil
+			}
+		}
+
+		if gatewayNodesToDeploy > 0 {
+			reporter.Failed(fmt.Errorf("there are insufficient nodes to deploy the required number of gateways"))
+			return nil
+		}
+	}
+
+	return nil
+}
+
+func (d *ocpGatewayDeployer) Cleanup(reporter api.Reporter) error {
+	reporter.Started("Removing the gateway configuration from nodes ")
+
+	computeClient, err := openstack.NewComputeV2(d.Client, gophercloud.EndpointOpts{Region: d.Region})
+	if err != nil {
+		return errors.WithMessagef(err, "error creating the compute client for the region: %q", d.Region)
+	}
+
+	gwNodesList, err := d.K8sClient.ListGatewayNodes()
+	if err != nil {
+		reporter.Failed(err)
+		return errors.WithMessage(err, "error listing the gateway nodes")
+	}
+
+	groupName := d.InfraID + gwSecurityGroupSuffix
+	gwNodes := gwNodesList.Items
+
+	for i := range gwNodes {
+		err = d.removeGWFirewallRules(groupName, gwNodes[i].Name, computeClient)
+		if err != nil {
+			reporter.Failed(err)
+			return errors.WithMessage(err, "error deleting the gateway secutiy group rules")
+		}
+	}
+
+	err = d.K8sClient.RemoveGWLabelFromWorkerNodes()
+	if err != nil {
+		reporter.Failed(err)
+		return errors.WithMessage(err, "failed to remove labels from worker node")
+	}
+
+	reporter.Succeeded("Successfully removed the gateway configuration from the nodes")
+
+	reporter.Started("Retrieving the Submariner gateway firewall rules")
+
+	err = d.deleteSG(groupName, computeClient)
+	if err != nil {
+		return errors.WithMessage(err, "error deleting the gateway security group")
+	}
+
+	reporter.Succeeded("Successfully deleted the g/w firewall rules")
```

Comment:
We don't abbreviate  elsewhere

```suggestion
	reporter.Succeeded("Successfully deleted the Submariner gateway firewall rules")
```